Domains: Update Domain Management contact information section

rafaelgallani opened the pull request:
#### Changes proposed in this Pull Request
This PR creates the new redesigned contact information component, shown in the domain settings page for registered and connected domains. This is part of the domain management pages redesign project detailed in pcYYhz-m2-p2.

#### Preview
![image](https://user-images.githubusercontent.com/18705930/146291410-f8e1d704-99c9-43fc-80c2-873e2fd415eb.png)
![image](https://user-images.githubusercontent.com/18705930/146291532-d9addfe0-babf-4842-9323-bd4906db85e9.png)
![image](https://user-images.githubusercontent.com/18705930/146294879-dc6e592c-d2d7-4b89-bcba-7234180c4901.png)


#### Testing instructions
- Build this branch locally or open the live Calypso link
- If you're in the live Calypso link, please append `?flags=domains/settings-page-redesign` to your URL to enable the appropriate feature flag
- Go to "Upgrades > Domains"
- Select one of your registered or mapped domains
- Ensure that:
  - The contact information card in the domain settings page looks like the screenshot above;
  - The privacy protection toggle works like in the old contact information page;
  - The privacy check (whether privacy is enabled or not) works the same as in the old contact information page;
  - The non-owner notice in the screenshot above is displayed when you're not the domain owner;
  - The correct information is displayed;
  - The links work correctly;
  - The "Edit" button takes you to the edit contact information page;
  - The "Manage consent" button takes you to the manage consent for personal data use page;
  - Also check that the mobile view looks correct.
  
  In short, make sure it matches the redesigned version in hlVh2q24ad6MCwlwNnE9PQ-fi-600%3A54818.

rafaelgallani committed:
Create component and styles

rafaelgallani committed:
Add toggle styles

rafaelgallani committed:
Cleanup

rafaelgallani committed:
Style tweak

rafaelgallani committed:
Add whois types

rafaelgallani committed:
Add whois information to accordion component

a8ci18n commented on client/my-sites/domains/domain-management/contacts-privacy/contacts-card.jsx:
```diff
@@ -0,0 +1,163 @@
+		const { translate, isUpdatingPrivacy } = this.props;
+
+		const label = privateDomain
+			? translate( 'Privacy protection on' )
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Privacy protection: on' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

a8ci18n commented on client/my-sites/domains/domain-management/contacts-privacy/contacts-card.jsx:
```diff
@@ -0,0 +1,163 @@
+
+		const label = privateDomain
+			? translate( 'Privacy protection on' )
+			: translate( 'Privacy protection off' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Privacy protection: off' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

a8ci18n commented on client/my-sites/domains/domain-management/settings/index.tsx:
```diff
@@ -41,6 +45,35 @@ const Settings = ( props: SettingsPageProps ): JSX.Element => {
+
+		const contactInfoFullName = `${ contactInformation.fname } ${ contactInformation.lname }`;
+		const privacyProtectionLabel = privateDomain
+			? translate( 'privacy protection on' )
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Privacy protection: on' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

a8ci18n commented on client/my-sites/domains/domain-management/settings/index.tsx:
```diff
@@ -41,6 +45,35 @@ const Settings = ( props: SettingsPageProps ): JSX.Element => {
+		const contactInfoFullName = `${ contactInformation.fname } ${ contactInformation.lname }`;
+		const privacyProtectionLabel = privateDomain
+			? translate( 'privacy protection on' )
+			: translate( 'privacy protection off' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Privacy protection: off' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

rafaelgallani committed:
Move logic to parent component

Logic moved to the parent component so it's possible to check whether the user is the actual domain owner and render a different component.

rafaelgallani committed:
Use placeholder when loading `whoisData`

rafaelgallani committed:
Fix manage consent button layout

rafaelgallani committed:
TS conversion

rafaelgallani committed:
FC conversion

a8ci18n commented on client/my-sites/domains/domain-management/contacts-privacy/contacts-card.tsx:
```diff
@@ -0,0 +1,168 @@
+		const { isUpdatingPrivacy } = props;
+
+		const label = privateDomain
+			? translate( 'Privacy protection on' )
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Privacy protection: on' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

a8ci18n commented on client/my-sites/domains/domain-management/contacts-privacy/contacts-card.tsx:
```diff
@@ -0,0 +1,168 @@
+
+		const label = privateDomain
+			? translate( 'Privacy protection on' )
+			: translate( 'Privacy protection off' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Privacy protection: off' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

rafaelgallani committed:
Include privacy protection link and add href to contacts component

rafaelgallani committed:
Add redesigned non-owner notice

rafaelgallani committed:
Show recommendation text only when privacy is available

rafaelgallani committed:
Restrict render only for registered domains

leonardost commented on client/my-sites/domains/domain-management/settings/index.tsx:
```diff
@@ -56,6 +61,53 @@ const Settings = ( props: SettingsPageProps ): JSX.Element => {
 		</>
 	);
 
+	const getContactInformationAccordion = () => {
```
I think there might be a problem here when viewing domains of which we are not owners - the `requestWhois` query will return an error, and the component's loading state will never be resolved. A quick and dirty fix would be adding a condition at the start of this method to check if the current user can manage this domain, something like

```
if ( ! domain.currentUserCanManage ) {
	return (
		<Accordion title="Contact information">
			<ContactsPrivacyInfo
				domains={ props.domains! }
				selectedSite={ props.selectedSite }
				selectedDomainName={ props.selectedDomainName }
			></ContactsPrivacyInfo>
		</Accordion>
	);
}
```

This duplicates some code, but I couldn't think of anything more elegant at the moment 😅

leonardost reviewed (changes requested):
This is looking great, nice job! 👍  I tested with some different domains and found only one issue when viewing a domain where I'm not the owner - the loading state never resolves because the `whois` query returns an error. I've left a fix suggestion in the code. Other than that, works great! 😄

rafaelgallani committed:
Move render function to `renderMainContent()`

rafaelgallani committed:
`getContactInformationAccordion()` -> `renderContactInformationSecion()`

rafaelgallani committed:
Move files to new card folder structure and revert previous changes

rafaelgalani commented:
Changing base as the folder structure/methods usage in #59334 is more adequate - also to avoid a future conflict resolution when rebasing 😄

a8ci18n commented on client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx:
```diff
@@ -0,0 +1,188 @@
+		const { isUpdatingPrivacy } = props;
+
+		const label = privateDomain
+			? translate( 'Privacy protection on' )
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Privacy protection: on' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

a8ci18n commented on client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx:
```diff
@@ -0,0 +1,188 @@
+
+		const label = privateDomain
+			? translate( 'Privacy protection on' )
+			: translate( 'Privacy protection off' );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Privacy protection: off' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

rafaelgallani committed:
Fix error where card wouldn't load if the user wasn't the owner

a8ci18n commented on client/my-sites/domains/domain-management/settings/index.tsx:
```diff
@@ -129,12 +136,71 @@ const Settings = ( {
+
+		const { privateDomain } = domain;
+		const privacyProtectionLabel = privateDomain
+			? translate( 'Privacy protection on', { textOnly: true } )
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'Privacy protection: on' )` *ES Score: 10*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

a8ci18n commented on client/my-sites/domains/domain-management/settings/index.tsx:
```diff
@@ -129,12 +136,71 @@ const Settings = ( {
+		const { privateDomain } = domain;
+		const privacyProtectionLabel = privateDomain
+			? translate( 'Privacy protection on', { textOnly: true } )
+			: translate( 'Privacy protection off', { textOnly: true } );
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **21** times:
`translate( 'Privacy protection: off' )` *ES Score: 11*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/domain-management-contact-information)

gius80 commented on client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx:
```diff
@@ -0,0 +1,188 @@
+						'Privacy protection is not available due to the registry’s policies. {{a}}Learn more{{/a}}',
+						{
+							components: {
+								a: <a href={ PRIVACY_PROTECTION } />,
```
I noticed that previous component had` target="blank"` for this link, maybe we can maintain the same approach here.

```suggestion
								a: <a href={ PRIVACY_PROTECTION } target="_blank" rel="noopener noreferrer" />,
```

gius80 commented on client/my-sites/domains/domain-management/settings/cards/contact-information/contacts-card.tsx:
```diff
@@ -0,0 +1,188 @@
+			<p className="contact-information__toggle-item">
+				{ translate( 'We recommend keeping privacy protection on. {{a}}Learn more{{/a}}', {
+					components: {
+						a: <a href={ PUBLIC_VS_PRIVATE } />,
```
Same as https://github.com/Automattic/wp-calypso/pull/59194/files#r772218992

gius80 commented:
I re-checked, now everything looks ok! Good job! 🤙

I Left just a couple of nitpicky comments.

rafaelgallani committed:
Add `target="_blank"` to links

rafaelgallani committed:
Remove whitespaces from `style.scss`

rafaelgallani committed:
Bypass TS code inspection with spread operator

The current TS code inspection is listing the \`disabled\` prop in the \`ToggleControl\` component as invalid, even though it is defined at the component. Maybe the `\@types\` package is not reflecting the props defined on the component, so this commit fixes it.

rafaelgallani committed:
Add optional type and cast to bypass TS errors

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7052628

Thank you @rafaelgalani for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7052628).
